unbuffered.go

```diff
@@ -0,0 +1,116 @@
+package fasthttp
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+)
+
+type UnbufferedWriter interface {
+	Write(p []byte) (int, error)
+	WriteHeaders() (int, error)
+	Close() error
+}
+
+type UnbufferedWriterHttp1 struct {
+	writer            *bufio.Writer
+	ctx               *RequestCtx
+	bodyChunkStarted  bool
+	bodyLastChunkSent bool
+}
+
+var ErrNotUnbuffered = errors.New("not unbuffered")
+var ErrClosedUnbufferedWriter = errors.New("closed unbuffered writer")
+
+// Ensure UnbufferedWriterHttp1 implements UnbufferedWriter.
+var _ UnbufferedWriter = &UnbufferedWriterHttp1{}
+
+// NewUnbufferedWriter
+//
+// Object must be discarded when request is finished
+func NewUnbufferedWriter(ctx *RequestCtx) *UnbufferedWriterHttp1 {
+	writer := acquireWriter(ctx)
+	return &UnbufferedWriterHttp1{ctx: ctx, writer: writer}
+}
+
+func (uw *UnbufferedWriterHttp1) Write(p []byte) (int, error) {
+	if uw.writer == nil || uw.ctx == nil {
+		return 0, ErrClosedUnbufferedWriter
+	}
+
+	// Write headers if not already sent
+	if !uw.ctx.Response.headersWritten {
+		_, err := uw.WriteHeaders()
+		if err != nil {
+			return 0, fmt.Errorf("error writing headers: %w", err)
+		}
+	}
+
+	// Write body. In chunks if content length is not set.
+	if uw.ctx.Response.Header.contentLength == -1 && uw.ctx.Response.Header.IsHTTP11() {
+		uw.bodyChunkStarted = true
+		err := writeChunk(uw.writer, p)
+		if err != nil {
+			return 0, err
+		}
+		uw.ctx.bytesSent += len(p) + 4 + countHexDigits(len(p))
+		return len(p), nil
+	}
+
+	n, err := uw.writer.Write(p)
+	uw.ctx.bytesSent += n
+
+	return n, err
+}
+
+func (uw *UnbufferedWriterHttp1) WriteHeaders() (int, error) {
+	if uw.writer == nil || uw.ctx == nil {
+		return 0, ErrClosedUnbufferedWriter
+	}
+
+	if !uw.ctx.Response.headersWritten {
+		if uw.ctx.Response.Header.contentLength == 0 && uw.ctx.Response.Header.IsHTTP11() {
+			if uw.ctx.Response.SkipBody {
+				uw.ctx.Response.Header.SetContentLength(0)
+			} else {
+				uw.ctx.Response.Header.SetContentLength(-1) // means Transfer-Encoding = chunked
+			}
+		}
+		h := uw.ctx.Response.Header.Header()
+		n, err := uw.writer.Write(h)
```

Comment:
Probably faster and more efficient to use [`uw.ctx.Response.Header.WriteTo(uw.writer)`](https://pkg.go.dev/github.com/valyala/fasthttp?utm_source=godoc#ResponseHeader.WriteTo)